bsmetadata/preprocessing_utils.py

```diff
@@ -241,39 +285,53 @@ def preprocess(self, examples: Dict[str, List]) -> Dict[str, List]:
                     "ent_desc": ent_desc,
                 }
                 example_metadata.append(en)
+
+        examples[self.col_to_store_metadata] = example_metadata_list
         return examples
 
 
 class GenerationLengthPreprocessor(MetadataPreprocessor):
     """An exemplary metadata preprocessor for adding generation length information based on text."""
 
-    def __init__(self, mode):
+    def __init__(
+        self,
+        mode,
+        col_to_store_metadata="metadata",
+        col_text="text",
+    ) -> None:
         # The length can be calculated for the whole text or for each sentence of a text individually.
         # We can specify a global length of a TEXT or a local length for each SENTENCE of a text.
         # Therefore, we provide two different modes: text (global) or sentence (local).
         self.mode = mode  # {text, sentence}
 
+        self.col_text = col_text
+        super().__init__(col_to_store_metadata=col_to_store_metadata)
+
     def preprocess(self, examples: Dict[str, List]) -> Dict[str, List]:
         """
         Iterate through all the examples retrieve the length meta information.
         """
-        for example_text, example_metadata in zip(examples["text"], examples["metadata"]):
+
+        example_metadata_list = (
+            examples[self.col_to_store_metadata]
+            if self.col_to_store_metadata in examples
+            else [[] for _ in range(len(examples[self.col_text]))]
+        )
+
+        for example_text, example_metadata in zip(examples[self.col_text], example_metadata_list):
             if self.mode == "text":
                 text_length = self._extract_length_from_text(example_text)
                 example_length = {"key": "length", "type": "global", "value": text_length}
+                if not example_length:
+                    continue
+                example_metadata.append(example_length)
             elif self.mode == "sentence":
                 example_length = self._extract_length_from_sentences(example_text)
+                example_metadata.extend(example_length)
             else:
                 print("Please select a valid length type [text or sentence].")
```

Comment:
@chkla , fyi, I've applied the changes proposed [here](https://github.com/bigscience-workshop/metadata/pull/81#discussion_r773284867), otherwise I've got an error with the index `0` in `m[0]`

Reply:
👍